### PR TITLE
Move to rspec

### PR DESCRIPTION
**Pull request by scharrier**
Switch from minitest to rspec as we have a better knowledge of it.
This will let use move faster now. We could go back to minitest later.

### CODE REVIEW

File: spec/bump/cli_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require "spec_helper"
+
+describe Bump::CLI do
+  around(:each) do |example|
+    modified_path = File.expand_path("../../..", __FILE__) + "/exe:" + ENV["PATH"]
+    ClimateControl.modify PATH: modified_path do
+      example.run
+    end
+  end
+end
```
Comment (mehlah) [inline spec/bump/cli_spec.rb]: Specs disappeared? 👻

Comment (scharrier) [reply spec/bump/cli_spec.rb]: Oops. Right. I could fully remove the file as the examples will be removed in the next PR, but let's be clean here: I re-add them now.

**Comment by scharrier** [comment]
@mehlah : fixed (I added the examples in specs). Is that :+1: for you?

File: spec/bump/cli_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    modified_path = File.expand_path("../../..", __FILE__) + "/exe:" + ENV["PATH"]
+    ClimateControl.modify PATH: modified_path do
+      example.run
+    end
+  end
+
+  describe "validate command" do
+    it "outputs the right thing" do
+      expect do
+        print `bump validate`
```
Comment (mehlah) [inline spec/bump/cli_spec.rb]: Is the `print` necessary? I don't think so as the matcher captures the stream https://github.com/rspec/rspec-expectations/blob/master/lib/rspec/matchers/built_in/output.rb

Comment (scharrier) [reply spec/bump/cli_spec.rb]: Yep, it is: `expected block to output "Deploying...\n" to stdout, but output nothing` without it.

**Pull request merged**